code/game/turfs/simulated/floor/plating/asteroid.dm

```diff
@@ -147,10 +89,10 @@
 		if("basalt5", "basalt9")
 			B.set_light(1.4, 0.6, LIGHT_COLOR_LAVA) //barely anything!
 
-/turf/open/floor/plating/asteroid/basalt/gets_dug()
-	if(!dug)
-		set_light(0)
+/turf/open/floor/plating/asteroid/basalt/ComponentActivated(datum/component/C)
 	..()
+	if(istype(C, /datum/component/archaeology))
+		set_light(0)
```

Comment:
I'm still concerned this will get called but not complete in the instant that someone hits it but doesn't finish the do_after

Reply:
You could return TRUE from the signal regardless of completing the do_after or not

Reply:
oh ok so it's working as intended awesome